src/controllers/AuthController.ts

```diff
@@ -0,0 +1,51 @@
+import AuthAPI from '../api/AuthAPI';
+import ChatController from './ChatController';
+import { router } from '../utils/router';
+import { errorHandler } from '../utils/errorHandler';
+import { ISignIn } from '../pages/SignIn/interfaces';
+import { ISignUp } from '../pages/SignUp/interfaces';
+import { Actions } from '../Store';
+
+class AuthController {
+    public async signIn(user: ISignIn) {
+        return AuthAPI.signIn(user)
+            .then(() => {
+                this.checkAuth()
+                    .then((response) => {
+                        Actions.setProfile(response);
+                        ChatController.request().then(() => {
+                            router.go('/messenger');
+                        }
+                        )
+                    })
+            })
+            .catch(errorHandler);
```

Comment:
Есть два подхода к работе с асинхронностью - async/await и promise-based. Они оба имеют место быть, но важно использовать что-то одно. Сейчас эти два подхода переплетаются, что создает путаницу в коде. Я бы предложил остановиться только на async/await, тогда выглядеть будет как-то так
```suggestion
    public async signIn(user: ISignIn) {
      try {
        await AuthAPI.signIn(user)
        const checkAuthResponse = await this.checkAuth()
        Actions.setProfile(checkAuthResponse)
        await ChatController.request();
        router.go('/messenger');
      } catch () {
      }
```

сразу убрали тройную вложенность. Так можно сделать во всех контроллерах